newIDE/app/src/ObjectGroupsList/index.js

```diff
@@ -231,18 +154,51 @@ export default class GroupsListContainer extends React.Component<Props, State> {
   _onEditName = (groupWithContext: GroupWithContext) => {
     this.setState(
       {
-        renamedGroupWithScope: groupWithContext,
+        renamedGroupWithContext: groupWithContext,
       },
-      () => this.sortableList.getWrappedInstance().forceUpdateGrid()
+      () => {
+        if (this.sortableList) this.sortableList.forceUpdateGrid();
+      }
+    );
+  };
+
+  _onDuplicate = (groupWithContext: GroupWithContext): ?GroupWithContext => {
+    const { group, global } = groupWithContext;
+    const { globalObjectGroups, objectGroups } = this.props;
+
+    const container: gdObjectGroupsContainer = global
+      ? globalObjectGroups
+      : objectGroups;
+
+    const newName = newNameGenerator(
+      group.getName(),
+      name => container.has(name),
+      ''
+    );
+
+    const newGroup = container.insertNew(
+      newName,
+      container.getPosition(group.getName()) + 1
+    );
+
+    unserializeFromJSObject(
+      newGroup,
+      serializeToJSObject(group),
+      'unserializeFrom'
     );
```

Comment:
It's probably the way it's done in other places in the code, but I find it strange to rely on serialization to copy the attributes of the group. Shouldn't containers have a `duplicate` method instead?

Reply:
I guess so but it kinda grows the scope of this PR 😅 
It's an improvement we could do later maybe? Wait for another time we need a duplicate action?